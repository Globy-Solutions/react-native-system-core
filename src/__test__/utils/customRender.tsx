import {render} from '@testing-library/react-native';
import {ErrorBoundary} from 'react-error-boundary';
import renderer from 'react-test-renderer';
import ThemeProvider from 'src/theme/theme-provider';

type Props = {
  children: React.ReactNode;
};

const logError = (error: Error, info: {componentStack: string}) => {
  // Do something with the error, e.g. log to an external API
  console.log(error, info);
};
const Provider = ({children}: Props) => (
  // @ts-ignore
  <ErrorBoundary FallbackComponent="Something went wrong" onError={logError}>
    <ThemeProvider>{children}</ThemeProvider>
  </ErrorBoundary>
);
const App = ({children}: Props) => (
  // @ts-ignore
  <ErrorBoundary FallbackComponent="Something went wrong" onError={logError}>
    {children}
  </ErrorBoundary>
);
const toJSON = (component: any) => {
  const r = App(component);
  renderer.create(r).toJSON();
  return render(r);
};

export * from '@testing-library/react-native';
export {App, Provider, toJSON};
