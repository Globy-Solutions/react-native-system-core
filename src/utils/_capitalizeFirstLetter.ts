export const capitalize = (text: string) => {
  const firstLetter = text.charAt(0);
  const firstLetterCap = firstLetter.toUpperCase();
  const remainingLetters = text.slice(1);
  return firstLetterCap + remainingLetters;
};

const capitalizeFirstLetter = (text: string) => {
  const textArr = /\b/i.test(text);
  if (textArr) {
    const arr = text.split(' ');
    return [arr[0], arr[1]]
      .map(t => capitalize(t as string).charAt(0))
      .join('');
  } else {
    return capitalize(text);
  }
};

export default capitalizeFirstLetter;
