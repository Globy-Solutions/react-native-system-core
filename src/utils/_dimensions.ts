import {Dimensions, PixelRatio} from 'react-native';

import type {DimensionValue} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

const {height, width}: {height?: DimensionValue; width?: DimensionValue} =
  Dimensions.get('screen');

const LINE_HEIGHT_SCALE_FACTOR = 1.33;

/**
 * Converts provided width percentage to independent pixel (dp).
 * @param  {string} widthPercent The percentage of screen's width that UI element should cover
 *                               along with the percentage symbol (%).
 * @return {number}              The calculated dp depending on current device's screen width.
 */
const widthPercentageToDP = (widthPercent: string | number) => {
  // Parse string percentage input and convert it to number.
  const elemWidth =
    typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent);
  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((width * elemWidth) / 100);
};

/**
 * Converts provided height percentage to independent pixel (dp).
 * @param  {string} heightPercent The percentage of screen's height that UI element should cover
 *                                along with the percentage symbol (%).
 * @return {number}               The calculated dp depending on current device's screen height.
 */
const heightPercentageToDP = (heightPercent: string | number): number => {
  // Parse string percentage input and convert it to number.
  const elemHeight =
    typeof heightPercent === 'number'
      ? heightPercent
      : parseFloat(heightPercent);

  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((height * elemHeight) / 100);
};

/*
* Implement that piece of code to have orientation detection

Dimensions.get('screen').width < Dimensions.get('screen').height ? 'portrait' : 'landscape'
*/

const guidelineBaseWidth: number = 375;
const guidelineBaseHeight: number = 667;
const horizontalScale = (size: number) => (width / guidelineBaseWidth) * size;
const verticalScale = (size: number) => (height / guidelineBaseHeight) * size;
const moderateScale = (size: number, factor = 0.5) =>
  size + (horizontalScale(size) - size) * factor;

const getFontLineHeight = (fontSize: number) =>
  fontSize * LINE_HEIGHT_SCALE_FACTOR;

export {
  getFontLineHeight,
  height,
  horizontalScale,
  heightPercentageToDP as hp,
  moderateScale,
  verticalScale,
  width,
  widthPercentageToDP as wp
};
