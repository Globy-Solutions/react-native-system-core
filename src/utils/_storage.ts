/**
 * Method for LocalStorage
 * setData allways return Array object
 *
 * @function
 */

import {MMKV} from 'react-native-mmkv';

const AsyncStorage = new MMKV();

const getData = (name: string) => {
  const value = AsyncStorage.getString(name);
  return value ? JSON.parse(value) : null;
};

const setData = (name: string, data: any): void => {
  const currentState = getData(name);
  let temp: any[] = [];
  if (currentState) {
    temp = [...currentState, data];
  } else {
    temp = [data];
  }
  AsyncStorage.set(name, JSON.stringify(temp));
};

const deleteData = (name: string) => {
  AsyncStorage.delete(name);
};

export {deleteData, getData, setData};
