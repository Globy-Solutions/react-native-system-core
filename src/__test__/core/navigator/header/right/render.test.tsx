import { Text } from 'react-native';

import { Provider, render, screen } from 'src/__test__/utils/customRender';
import HeaderRight from 'src/navigator/header/right';


describe('HeaderRight - Render', () => {

  let headerRight: any;
  beforeAll(() => {
    const { getByTestId } = render(<Provider>
      <HeaderRight>
        <Text>Test Child</Text>
      </HeaderRight>
    </Provider>);
    headerRight = getByTestId('headerRight');
  });

  test('snapshot', () => {
    expect(screen).toMatchSnapshot();
  });
  test('renders children correctly', () => {
    expect(headerRight).toBeTruthy();
  });

});
