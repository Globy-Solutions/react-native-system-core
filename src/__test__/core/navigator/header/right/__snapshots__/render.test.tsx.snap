// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRight - Render snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginLeft": "auto",
        "maxHeight": 66.5,
        "minHeight": 66.5,
        "minWidth": 187.5,
      },
      {
        "backgroundColor": "#FFAF33",
        "marginRight": -10,
        "width": null,
      },
    ]
  }
  testID="headerRight"
>
  <Text>
    Test Child
  </Text>
</View>
`;
